entities/library.rb

```diff
@@ -0,0 +1,38 @@
+# described library class - get output information
+class Library
+  attr_accessor :data_module
+  def initialize(data_module)
+    @data_module = data_module
+  end
+
+  def most_active_reader
+    data_module.fetch_orders.group_by { |order| order.reader }.sort_by { |_, value| value.length }.to_a.last.first
+  end
+
+  def most_popular_book
+    group_orders_by_books.last.first
+  end
+
+  def three_popular_books_orders
+    books = group_orders_by_books.last(3).map { |item| item.first }
+    book_ids = books.map { |book| book.id }
+
+    readers = data_module.fetch_orders.map do |order|
+      if book_ids.include? order.book.id
+        order.reader
+      else
+        nil
+      end
+    end
```

Comment:
Now method is very long and are several extra actions
There are several possibilities here:
1. data_module.fetch_orders.select { |order| book_ids.include? order.book.id }.map(&:reader)
2. data_module.fetch_orders.map { |order| order.reader if  book_ids.include? order.book.id }.compact

According to task you should return count of uniq readers who takes at least one of 3 most popular books. Are you sure that you need to return books and readers.

Also you don't need to use return in the line 30